basictracer/span.py

```diff
@@ -21,7 +21,9 @@ def __init__(
             start_time=None):
         super(BasicSpan, self).__init__(tracer, context)
         self._tracer = tracer
-        self._lock = Lock()
+        self._refcount = 1
+        self._to_restore = self._tracer.active_span
```

Comment:
Hey, a small note after Ben's comment:

Marrying the `Span` creation to the active span source may not be an ideal thing. Having separated the logic for creation AND activation/deactivation could keep things separated and cleaner.

Moreover: with the current implementation, when we `capture` a Span, we need to *keep* track of the previously active `Span`, for *both* the calling function and the callback function (with the current approach, we *only* `capture` such active Span for the calling function).